libtenzir/include/tenzir/series.hpp

```diff
@@ -19,11 +20,15 @@
 namespace tenzir {
 
 /// A temporary series representation (until we have a proper one).
-struct typed_array {
-  typed_array() = default;
+struct series {
```

Comment:
Maybe this should be 

```cpp
template<class Type = type>
struct series { ... }
```

But we can also keep this as-is for now. We will surely see how we want to do it when we build the evaluator.

Reply:
I tried that. The problem is that we need to assign a type ID to id so that we can send it with CAF, and that requires being able to forward-declare it. It's a mess from a code-organization perspective that I'd like to avoid for now.